core/nandtool/flashmng.c

```diff
@@ -57,6 +67,23 @@ int flashmng_writeraw(oid_t oid, unsigned int page, const void *data, size_t siz
 }
 
 
+static int writemeta(oid_t oid, uint32_t addr, const void *data, size_t size)
+{
+	return write_ex(oid, addr, data, size, flashsrv_devctl_writemeta);
+}
+
+
+int flashmng_writeraw(oid_t oid, unsigned int page, const void *data, size_t size)
+{
+	flashsrv_info_t *info = flashmng_info(oid);
+
+	if (info == NULL)
+		return -EFAULT;
```

Comment:
I would say `-EINVAL,` `-EFAULT` is pretty much reserved for kernel to say that user tried to pass it some nasty pointer

Reply:
all other `flashmng` functions return `EFAULT` in this case. I would personally go for `EIO` as inability to read flash info suggests someone tried to use nandwrite on non-NAND device.

Leaving as-is for compatibility

Reply:
@Maxez  this is something worth to be done as a separate PR

Reply:
I'll change `EFAULT` usage to `EIO` in another PR.